From da17279076919f55385adc2ef71dd7ac90596380 Mon Sep 17 00:00:00 2001
From: Roger Quadros <rogerq@ti.com>
Date: Thu, 22 Nov 2018 14:13:03 +0200
Subject: [PATCH] examples:lab_5:am335x: Update Interrupt map resource

Use new format of Vendor specific Interrupt map resource.

Signed-off-by: Roger Quadros <rogerq@ti.com>
[cristi: ported to pru-software-support-package v5.7.0]
Signed-off-by: Cristian Ciocaltea <cristian.ciocaltea@gmail.com>
---
 .../resource_table_1.h                        | 44 +++++++++----------
 1 file changed, 21 insertions(+), 23 deletions(-)

diff --git a/labs/Hands_on_Labs/lab_5/solution/am335x/PRU_RPMsg_Echo_Interrupt1/resource_table_1.h b/labs/Hands_on_Labs/lab_5/solution/am335x/PRU_RPMsg_Echo_Interrupt1/resource_table_1.h
index 9e18d52..0dee9a0 100644
--- a/labs/Hands_on_Labs/lab_5/solution/am335x/PRU_RPMsg_Echo_Interrupt1/resource_table_1.h
+++ b/labs/Hands_on_Labs/lab_5/solution/am335x/PRU_RPMsg_Echo_Interrupt1/resource_table_1.h
@@ -56,10 +56,7 @@
 /* Definition for unused interrupts */
 #define HOST_UNUSED		255
 
-/* Mapping sysevts to a channel. Each pair contains a sysevt, channel. */
-struct ch_map pru_intc_map[] = { {18, 3},
-				 {19, 1},
-};
+#define NUM_INTR_MAPS		2
 
 struct my_resource_table {
 	struct resource_table base;
@@ -71,8 +68,11 @@ struct my_resource_table {
 	struct fw_rsc_vdev_vring rpmsg_vring0;
 	struct fw_rsc_vdev_vring rpmsg_vring1;
 
-	/* intc definition */
-	struct fw_rsc_custom pru_ints;
+        /* vendor resource header */
+        struct fw_rsc_vendor ven_hdr;
+        /* intrmap resource data */
+        struct fw_rsc_pruss_intrmap_hdr intr_hdr;
+        struct fw_rsc_pruss_intrmap_data intr_data[NUM_INTR_MAPS];
 };
 
 #pragma DATA_SECTION(resourceTable, ".resource_table")
@@ -84,7 +84,7 @@ struct my_resource_table resourceTable = {
 	/* offsets to entries */
 	{
 		offsetof(struct my_resource_table, rpmsg_vdev),
-		offsetof(struct my_resource_table, pru_ints),
+		offsetof(struct my_resource_table, ven_hdr),
 	},
 
 	/* rpmsg vdev entry */
@@ -116,20 +116,18 @@ struct my_resource_table resourceTable =
 		0                       //reserved
 	},
 
-	{
-		TYPE_POSTLOAD_VENDOR, PRU_INTS_VER0 | TYPE_PRU_INTS,
-		sizeof(struct fw_rsc_custom_ints),
-		{
-			0x0000,
-			/* Channel-to-host mapping, 255 for unused */
-			HOST_UNUSED, 1, HOST_UNUSED, 3, HOST_UNUSED,
-			HOST_UNUSED, HOST_UNUSED, HOST_UNUSED, HOST_UNUSED, HOST_UNUSED,
-			/* Number of evts being mapped to channels */
-			(sizeof(pru_intc_map) / sizeof(struct ch_map)),
-			/* Pointer to the structure containing mapped events */
-			pru_intc_map,
-		},
-	},
+        {       /* Vendor specific resource */
+                TYPE_VENDOR,
+        },
+        {       /* INTR map Header */
+                TYPE_PRU_INTS,  /* INTRMAP type */
+                1,              /* version */
+                NUM_INTR_MAPS,  /* num_maps */
+        },
+        {       /* INTR map data */
+                { 19, 1, 1 },
+                { 18, 3, 3 },
+        },
 };
 
 #endif /* _RSC_TABLE_PRU_H_ */
